File: amqp/src/test/java/edu/upc/ichnaea/amqp/model/DatasetTest.java
package edu.upc.ichnaea.amqp.model;

import org.junit.*;
import static org.junit.Assert.*;

public class DatasetTest {

    @Test
    public void testReadingRows()
    {
        Dataset set = new Dataset();
        DatasetColumn column = new DatasetColumn("test");
        column.add(1);
        column.add(2);
        column.add(3);
        set.add(column);
        DatasetRow row = set.getRow(0);
        
        assertEquals(1, row.get("test").intValue());
        assertEquals(1, row.size());
        
        column = new DatasetColumn("test2");
        
        column.add(4);
        column.add(5);
        
        set.add(column);
        
        row = set.getRow(0);
        
        assertEquals(1, row.get("test").intValue());
        assertEquals(4, row.get("test2").intValue());        
        assertEquals(2, row.size()); 
        
        row = set.getRow(2);
        
        assertEquals(3, row.get("test").intValue());
        assertEquals(1, row.size());
        
        column = new DatasetColumn("test3");
        
        column.add(6);
        column.add(7);
        column.add(8);
        
        set.add(column);        

        row = set.getRow(2);
        
        assertEquals(3, row.get("test").intValue());
        assertEquals(8, row.get("test3").intValue());        
        assertEquals(2, row.size());
        
    }

}
    

File: amqp/php/src/Ichnaea/Amqp/Tests/Model/DatasetTest.php
<?php

namespace Ichnaea\Amqp\Tests\Model;

use Ichnaea\Amqp\Model\Dataset;

class DatasetTest extends \PHPUnit_Framework_TestCase
{
    public function testSettingColumns()
    {
        $data = new Dataset();
        $data->setColumns(array(
            "BA"    => array(1, 2, 3, 4, 5),
            "CNFC"  => array(1, 2, 3, 5, 6),
        ));

        $this->assertEquals(array("BA", "CNFC"), $data->getColumnNames(), "Column names are the keys of the columns array.");
        $this->assertEquals(array(1, 2, 3, 4, 5), $data->getColumn("BA"), "Columns return their row values by name.");
    }

    /**
     * @expectedException InvalidArgumentException
     */
    public function testSettingInvalidColumn()
    {
        $data = new Dataset(array("BA"=>5));
    }

    public function testReadingCsv()
    {
        $data = new Dataset("BA; CNFC
1; 1
2; 2
3; 3
4; 5
5; 6

");

        $this->assertEquals(array("BA", "CNFC"), $data->getColumnNames(), "Column names are the keys of the columns array.");
        $this->assertEquals(array(1, 2, 3, 4, 5), $data->getColumn("BA"), "Columns return their row values by name.");
        $this->assertEquals(array(1, 2, 3, 5, 6), $data->getColumn("CNFC"), "Columns return their row values by name.");
    }

    public function testGettingRows()
    {
        $data = new Dataset();
        $data->setColumns(array(
            "BA"    => array(1, 2, 3),
            "CNFC"  => array(1, 2, 4),
        ));
        $rows = $data->getRows();
        $this->assertEquals(array(array(1, 1), array(2, 2), array(3, 4)), $rows, "Dataset returns rows correctly.");   
    }

    public function testToString()
    {
        $data = new Dataset();
        $data->setColumns(array(
            "BA"    => array(1, 2, 3),
            "CNFC"  => array(1, 2, 4),
        ));
        $str = $data->toString(";","'");
        $expect = "'BA';'CNFC'\n'1';'1'\n'2';'2'\n'3';'4'";
        $this->assertEquals($expect, $str, "Dataset to string works correctly.");
    }

}
